File: app/src/main/java/com/example/fitnesstracker/Step5Fragment.java
package com.example.fitnesstracker;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.google.android.material.card.MaterialCardView;

public class Step5Fragment extends Fragment {
    MaterialCardView cardBeginner, cardIntermediate, cardAdvance;
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_step5, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        cardBeginner = view.findViewById(R.id.cardBeginner);
        cardIntermediate = view.findViewById(R.id.cardIntermediate);
        cardAdvance = view.findViewById(R.id.cardAdvance);
        cardBeginner.setOnClickListener(v -> selectExperience(cardBeginner));
        cardIntermediate.setOnClickListener(v -> selectExperience(cardIntermediate));
        cardAdvance.setOnClickListener(v -> selectExperience(cardAdvance));
    }

    private void selectExperience(MaterialCardView selectedCard) {
        cardBeginner.setChecked(false);
        cardIntermediate.setChecked(false);
        cardAdvance.setChecked(false);
        selectedCard.setChecked(true);
    }
}

File: app/src/main/java/com/example/fitnesstracker/view/fragments/Step5Fragment.kt
package com.example.fitnesstracker.view.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.fitnesstracker.R
import com.google.android.material.card.MaterialCardView

class Step5Fragment : Fragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_step5, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        cardBeginner = view.findViewById(R.id.cardBeginner)
        cardIntermediate = view.findViewById(R.id.cardIntermediate)
        cardAdvance = view.findViewById(R.id.cardAdvance)
        cardBeginner?.setOnClickListener { v: View? ->
            selectExperience(
                cardBeginner!!
            )
        }
        cardIntermediate?.setOnClickListener { v: View? ->
            selectExperience(
                cardIntermediate!!
            )
        }
        cardAdvance?.setOnClickListener { v: View? ->
            selectExperience(
                cardAdvance!!
            )
        }
    }

    private fun selectExperience(selectedCard: MaterialCardView) {
        cardBeginner?.isChecked = false
        cardIntermediate?.isChecked = false
        cardAdvance?.isChecked = false
        selectedCard.isChecked = true
    }

    companion object {
        private var cardBeginner: MaterialCardView? = null
        private var cardIntermediate: MaterialCardView? = null
        private var cardAdvance: MaterialCardView? = null

        val experience: String
            get() {
                var experience = ""
                if (cardBeginner!!.isChecked) {
                    experience = "Beginner"
                } else if (cardIntermediate!!.isChecked) {
                    experience = "Intermediate"
                } else if (cardAdvance!!.isChecked) {
                    experience = "Advance"
                }
                return experience
            }
    }
}
